### PR TITLE
Upgrade from JBuilder to Dune and document building and testing instructions

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

**Comment by gasche** [comment]
(When I saw the PR title I didn't get it, as we have migrate to dune already. The point is to move from "jbuilder" to "dune", so rather an upgrade than yet another build system change.)

**Comment by anmonteiro** [comment]
> (When I saw the PR title I didn't get it, as we have migrate to dune already. The point is to move from "jbuilder" to "dune", so rather an upgrade than yet another build system change.)

That's correct. I'll update the title accordingly once I squash my commits.

**Comment by gasche** [comment]
(There is some discussion of this PR at https://github.com/ocaml-ppx/ppx_deriving_yojson/issues/83#issuecomment-446245657 )

Again: thanks a lot, this is great. You offered to create Changelog entries as well (feel free to say if this is too much work); let me know if/when this is ready, and I'll finish the release.

**Comment by anmonteiro** [comment]
@gasche I added a commit adding the CHANGELOG entries just now, but I tagged it as `Unreleased`. Should I change to 3.2 instead?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+---
+
+  * Add `let _ = to_yojson / of_yojson` to generated code to avoid warnings when
+    they aren't used
+    (#68)
+    @stevebleazard
+  * Fix bug where doing [@@deriving of_yojson] causes an unused rec warning
+    (#68)
+    @stevebleazard
+  * Add generated `ty_of_yojson_exn` raise an exception rather than return an
```
Comment (gasche) [inline CHANGELOG.md]: +to raise an exception

**Comment by gasche** [comment]
This should be good to go. (After I merge, I will update the Changelog to use people's Real Name when they have a preferred one, instead of github handles.) You seem to be thinking of rebasing to squash commits, let me know when you agree this is ready for merging.

**Comment by anmonteiro** [comment]
@gasche Done! Thanks for your review.

**Pull request merged**